src/sentry/interfaces/exception.py

```diff
@@ -795,6 +796,46 @@ def iter_tags(self):
             yield ('handled', self.handled and 'yes' or 'no')
 
 
+def uncontribute_non_stacktrace_variants(variants):
+    """If we have multiple variants and at least one has a stacktrace, we
+    want to mark all non stacktrace variants non contributing.  The reason
+    for this is that otherwise we end up in very generic grouping which has
+    some negative consequences for the quality of the groups.
+    """
+    if len(variants) <= 1:
+        return variants
+    any_stacktrace_contributes = False
+    non_contributing_variants = []
+    stacktrace_variants = set()
```

Comment:
Kind of confusing how `non_contributing_variants` contains components but `stacktrace_variants` contains variant names, this naming could probably be improved?